extension/googleclientauthextension/factory.go

```diff
@@ -27,6 +28,10 @@ import (
 	"google.golang.org/grpc/credentials/oauth"
 )
 
+const (
+	envProjectID = "GOOGLE_CLOUD_PROJECT"
```

Comment:
Can you add a comment linking to documentation, or code that also uses this?  I.e. "Why did we choose this env var?"